lib/cisco_node_utils/ace_v4.rb

```diff
@@ -0,0 +1,147 @@
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+
+module Cisco
+  # RouterAcl - node utility class for RouterAcl config mgmt.
+  class RouterAce < NodeUtil
+    attr_reader :acl_name, :seqno, :action, :proto, :v4_src_addr_format, :v4_dst_addr_format
+    attr_reader :v4_src_port_format, :v4_dst_port_format, :option_format
+
+    # name: name of the router instance
+    # instantiate: true = create router instance
+    def initialize(acl_name, seqno, action, proto, v4_src_addr_format,
+                   v4_dst_addr_format, v4_src_port_format, v4_dst_port_format,
+                   option_format='', instantiate=true)
+      addr_regex = /(any|host \d+\.\d+\.\d+\.\d+|\d+\.\d+\.\d+\.\d+\/\d+|\d+\.\d+\.\d+\.\d+ \d+\.\d+\.\d+\.\d+)/
+      addr_port_regex = /((eq|neq|lt|gt)\s+\d+|range\s+\d+\s+\d+){0,1}/
+      action_regex = /(permit|deny)/
+      proto_regex = /(sctp|tcp|ip|udp|pim|icmp|igmp|gre|nos|pcp|ospf|eigrp|ahp|esp|ipv6)/
+      fail ArgumentError.new("acl name len must be greater than 0") unless acl_name.length > 0
+      fail ArgumentError.new("seqno must be an integer") unless seqno.is_a?(Integer)
+      fail ArgumentError.new("action has to be permit or deny") unless action_regex.match(action)
+      fail ArgumentError.new("proto is invalid") unless proto_regex.match(proto)
+      fail ArgumentError.new("invalid src addr #{v4_src_addr_format}") unless addr_regex.match(v4_src_addr_format)
```

Comment:
Probably need to be consistent in your error messages and include the value of the item you are processing.  This is only done for #{v4_src_addr_format}.